app/models/membership.rb

```diff
@@ -1,7 +1,30 @@
 class Membership < ActiveRecord::Base
-  belongs_to :profile
   belongs_to :organisation
+  belongs_to :user
 
-  validates_uniqueness_of :profile_id, scope: :organisation_id
-  validates_presence_of :profile, :organisation
+  store_accessor :permissions, :roles, :applications
+
+  scope :with_roles, ->(*roles) {
+    where("permissions -> 'roles' ?& array['#{Array(roles).join("','")}']")
+  }
+
+  scope :with_any_role, ->(*roles) {
+    where("permissions -> 'roles' ?| array['#{Array(roles).join("','")}']")
+  }
```

Comment:
So I've used Postgres Array somewhere the other day and you can do this:

```Ruby
where("permissions -> 'roles' ?| ARRAY[?]", roles)
```

And ActiveRecord will handle that for you.